File: Languages/Kotlin/Day013.kt
// Languages - Kotlin - Day 13
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that takes an array of numbers, calculates the sum of all elements, and returns the result. Call this function with a sample array and print the result.

fun sumOfElements(arr: List<Int>): Int {
    // Calculate the sum of the array elements
    var sum = 0
    for (num in arr) {
        sum += num
    }
    return sum
}

fun main() {
    // Ask the user for the number of elements
    println("Enter the number of elements in the array:")
    val n = readLine()?.toIntOrNull()

    if (n == null || n <= 0) {
        println("Please enter a valid positive integer for the number of elements.")
        return
    }

    // Read the elements of the array
    val arr = mutableListOf<Int>()
    for (i in 1..n) {
        println("Enter element $i:")
        val input = readLine()?.toIntOrNull()
        if (input != null) {
            arr.add(input)
        } else {
            println("Invalid input. Please enter a valid integer.")
            return
        }
    }

    // Calculate and display the sum of the array
    val result = sumOfElements(arr)
    println("The sum of the array elements is: $result")
}

File: Languages/C++/Day013.cpp
// Languages - C++ - Day 13
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that takes an array of numbers, calculates the sum of all elements, and returns the result. Call this function with a sample array and print the result.

#include <iostream>
#include <vector>
#include <limits>

int sumArray(const std::vector<int> &arr)
{
    int sum = 0;
    for (int num : arr)
    {
        sum += num;
    }
    return sum;
}

int main()
{
    int n;

    // Prompt the user for the number of elements
    std::cout << "Enter the number of elements in the array: ";
    while (!(std::cin >> n) || n <= 0)
    {
        std::cout << "Please enter a valid positive integer: ";
        std::cin.clear();
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    }

    // Read the elements of the array
    std::vector<int> arr;
    std::cout << "Enter " << n << " integers:\n";
    for (int i = 0; i < n; i++)
    {
        int element;
        while (!(std::cin >> element))
        {
            std::cout << "Invalid input. Please enter an integer: ";
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
        }
        arr.push_back(element);
    }

    // Calculate and display the sum of the array
    int result = sumArray(arr);
    std::cout << "The sum of the array is: " << result << std::endl;

    return 0;
}
